### PR TITLE
Update networking section of components overview

### LINKED ISSUE #7066: Network section of docs/components.md is outdated
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind enhancement

**What happened**:

The [section](https://github.com/kubevirt/kubevirt/blob/main/docs/components.md#networking) shares future plans on how advanced networking should be done in KubeVirt. Since it was written, we have advanced in this area and that should be reflected there.

**What you expected to happen**:

The section should link to the [cluster-network-addons-operator](https://github.com/kubevirt/cluster-network-addons-operator) and describe how we envision advanced network scenarios with KubeVirt.

### PR DESCRIPTION
**Pull request by phoracek**
**What this PR does / why we need it**:

The section shares future plans on how advanced networking should be
done in KubeVirt. Since it was written, we have advanced in this area
and that should be reflected there.

This patch updates it to reflect the current architecture and focus.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #7066

**Special notes for your reviewer**:

It does not mention any specific plugins or extensions on purpose, so
the section does not become outdated with future integrations or bugs.
We should track details about APIs and extensions in the user-guide
instead and keep this document high-level.

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: docs/components.md [hunk 1]
```diff
@@ -185,15 +185,24 @@ However KubeVirt might define a Storage CRD along side with a flow description w
...
+KubeVirt aims to seamlessly integrate into Kubernetes networking, with VMs
+connected to the same network as Pods, using the same resources and same APIs.
+Without any external extensions, VMs in KubeVirt are capable of using the pod
+network, Services, or NetworkPolicies.
+
+To implement this, VMs are only bound to network resources that are made
+available to the virt-launcher Pod's network namespace. With this model, a wide
+variety of networking plugins can be supported.
+
+We also care for advanced networking scenarios, such as use of service meshes,
```
Comment (dankenigsberg) [inline docs/components.md:197]:
```suggestion
We also care for advanced networking scenarios, such as connecting to service meshes,
```
(I found it hard to parse the original)

Comment (phoracek) [reply docs/components.md]: Works for me. Fixed

**Review by dankenigsberg — APPROVED** [review]
